agent/engine/docker_task_engine.go

```diff
@@ -873,6 +873,12 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
 		return metadata
 	}
 	pullSucceeded := metadata.Error == nil
+	if pullSucceeded && engine.cfg.ContainerPullInParallel.Enabled() {
```

Comment:
I think the config var check is not needed now as now all pulled containers will be added to state

Reply:
this is from 1st commit in the PR, please review only the 2nd commit (details in PR description)

Reply:
oh yes...thanks